### PR TITLE
SEQNG-1072 Fixes to execute N&S in Epics

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds some fixes to successfully run N&S with epics, thanks @jluhrs 
The two main changes are

* Remove the call to set rows in between stages
* Handle the response of the observe pause differently if N&S

There are few other logger related changes

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala [hunk 1]
```diff
@@ -363,78 +374,81 @@ object GmosControllerEpics extends GmosEncoders {
...
-          sys.post.void
-
       override def observe(fileId: ImageFileId, expTime: Time): IO[ObserveCommandResult] =
         failOnDHSNotConected *>
           sys.observeCmd.setLabel(fileId) *>
           sys.observeCmd.setTimeout[IO](expTime + ReadoutTimeout) *>
-          sys.observeCmd.post[IO]
+          sys.observeCmd.post[IO].flatMap {
+            case ObserveCommandResult.Paused =>
+              sys.nsState.map(GmosEncoders.nsStateDecoder.decode).map {
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala]: @jluhrs This worked fine in test but there maybe a little risk given that GMOS is automagically changing this to `CLASSIC`. WDYT?

File: modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala [hunk 2]
```diff
@@ -363,78 +374,81 @@ object GmosControllerEpics extends GmosEncoders {
...
-          sys.configCmd.setTimeout[IO](ConfigTimeout) *>
-          sys.post.void
-
       override def observe(fileId: ImageFileId, expTime: Time): IO[ObserveCommandResult] =
         failOnDHSNotConected *>
           sys.observeCmd.setLabel(fileId) *>
           sys.observeCmd.setTimeout[IO](expTime + ReadoutTimeout) *>
-          sys.observeCmd.post[IO]
+          sys.observeCmd.post[IO].flatMap {
+            case ObserveCommandResult.Paused =>
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala]: Core change 1. We should respond with `Partial` if it is N&S to let the rest of the engine logic to proceed

File: modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosInstrumentActions.scala [hunk 3]
```diff
@@ -105,50 +105,37 @@ class GmosInstrumentActions[F[_]: MonadError[?[_], Throwable]: Concurrent: Logge
...
     env:       ObserveEnvironment[F],
     post:      (Stream[F, Result[F]], ObserveEnvironment[F]) => Stream[F, Result[F]]
   ): Stream[F, Result[F]] = {
-    val rowsToShuffle = Gmos.rowsToShuffle(sub.stage, rows)
     val nsPositionO   = positions.find(_.stage === sub.stage)
-    // Configure GMOS rows
-    Stream
-      .eval(
-        inst
-          .configureShuffle(rowsToShuffle)
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosInstrumentActions.scala]: Core change 2. We shouldn't set rows to shuffle in between subexposures. Discovered experimentally

File: modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala (hunk 1 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala]: Yes, but it changes to CLASSIC after the detector readout. At that point the observation state cannot change to PAUSED.

**Pull request merged**